README.md

```diff
@@ -179,6 +179,23 @@ Nightly builds from this repo are published to [MyGet](https://www.myget.org),
 and can be installed using the
 `https://www.myget.org/F/opentelemetry/api/v3/index.json` source.
 
+### Digital signing
+
+Starting with release
+[core-1.10.0-rc.1](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.10.0-rc.1),
+the DLLs included in the NuGet packages from this repository are digitally
+signed using [Sigstore](https://www.sigstore.dev/). Within each NuGet package,
+the digital signature and its corresponding certificate file are placed
+alongside the DLL, sharing the same name prefix, which ensures easy
+identification and association. This setup simplifies the process of signature
+verification, which can be carried out using the
+[cosign](https://github.com/sigstore/cosign) tool from Sigstore, as demonstrated
+in the example below, where a successful verification outputs "Verify OK".
+
+```bash
+cosign verify-blob --signature OpenTelemetry.dll-keyless.sig --certificate OpenTelemetry.dll-keyless.pem.cer OpenTelemetry.dll --certificate-identity "https://github.com/open-telemetry/opentelemetry-dotnet/.github/workflows/publish-packages-1.0.yml@refs/tags/core-1.10.0-rc.1" --certificate-oidc-issuer "https://token.actions.githubusercontent.com"
```

Comment:
@ThomsonTan 

Is it possible or no to change the identity of the cert?

Instead of...

`https://github.com/open-telemetry/opentelemetry-dotnet/.github/workflows/publish-packages-1.0.yml@refs/tags/core-1.10.0-rc.1`

Thinking something like...

`https://github.com/open-telemetry/opentelemetry-dotnet/core/1.10.0-rc.1`

...would be more palatable for anyone trying to automate verification.

Reply:
The default identity comes from the OIDC issuer `https://token.actions.githubusercontent.com/` to unique identify the workflow which requested the cert, while the suggested identity doesn't identify the workflow. I don't am not sure it can be customized. Probably file an separated issue on it?

BTW: if the customer doesn't want to do exact match on the default ID, there is an option [` --certificate-identity-regex`](https://github.com/sigstore/cosign/blob/main/doc/cosign_verify-attestation.md) which accepts a regex with wildcard match for the identity. Like `--certificate-identity-regex "https://github\.com/open-telemetry/opentelemetry-dotnet/.+/core-1\.10\.0-rc\.1".

Reply:
I think this is fine. Should we show the regex version too in the docs? Or maybe just add a link to it?

Reply:
Added a document link to the other verification options.